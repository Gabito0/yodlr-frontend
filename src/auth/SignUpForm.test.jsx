import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignupForm from "./SignUpForm";
import { vitest, expect, it } from "vitest";

// Mock the signup function
const mockSignup = vitest.fn();

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <SignupForm signup={mockSignup} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

it("submits form with valid data", async function () {
  mockSignup.mockResolvedValueOnce({ success: true });

  const { getByLabelText, getByText } = render(
    <MemoryRouter>
      <SignupForm signup={mockSignup} />
    </MemoryRouter>
  );

  fireEvent.change(getByLabelText("Email"), {
    target: { value: "newuser@example.com" },
  });
  fireEvent.change(getByLabelText("First name"), {
    target: { value: "New" },
  });
  fireEvent.change(getByLabelText("Last name"), {
    target: { value: "User" },
  });

  fireEvent.click(getByText("Submit"));

  await waitFor(() => {
    expect(mockSignup).toHaveBeenCalledWith({
      email: "newuser@example.com",
      firstName: "New",
      lastName: "User",
    });
  });
});

it("shows error messages on failed submission", async function () {
  mockSignup.mockResolvedValueOnce({
    success: false,
    errors: ["Email is already taken", "Password is too weak"],
  });

  const { getByLabelText, getByText, findByText } = render(
    <MemoryRouter>
      <SignupForm signup={mockSignup} />
    </MemoryRouter>
  );

  fireEvent.change(getByLabelText("Email"), {
    target: { value: "existinguser@example.com" },
  });
  fireEvent.change(getByLabelText("First name"), {
    target: { value: "Existing" },
  });
  fireEvent.change(getByLabelText("Last name"), {
    target: { value: "User" },
  });

  fireEvent.click(getByText("Submit"));

  expect(await findByText("Email is already taken")).toBeInTheDocument();
  expect(await findByText("Password is too weak")).toBeInTheDocument();
});
