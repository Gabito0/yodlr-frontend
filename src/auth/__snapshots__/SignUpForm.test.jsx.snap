// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        Sign Up
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="firstName"
              >
                First name
              </label>
              <input
                class="form-control"
                id="firstName"
                name="firstName"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="lastName"
              >
                Last name
              </label>
              <input
                class="form-control"
                id="lastName"
                name="lastName"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary float-right mt-3"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
